src/jit/flowgraph.cpp

```diff
@@ -5763,7 +5772,7 @@ void Compiler::fgFindBasicBlocks()
             impInlineInfo->iciCall->gtCallMoreFlags |= GTF_CALL_M_DOES_NOT_RETURN;
```

Comment:
Ah yeah its just adjusts jump targets for exception handlers, doesn't actually change any of the returns. Probably should move the observation up also then leave the assert.